### ISSUE TITLE
Jp2 in PDF : Insufficient data for an image

### ISSUE THREAD

**Issue by Tyrben** [uclouvain/openjpeg#1412]
## Expected behavior and actual behavior.
Including any valid jp2 data in a PDF images steam should give a valid PDF.
This seems the case for some sort of jpeg2000, but not for other.
The error is triggered by Acrobat Reader DC, so, this is fully a black box.

## Steps to reproduce the problem.
Create a PDF containing jpeg2000 images.
Flip a coin.
Read the error message: Insufficient data for an image.

## Operating system
Windows

## openjpeg version
2.4.0
(and 1.2.0)

**Comment by rouault** [comment]
There's no evidence the issue comes from openjpeg. Perhaps the way you create the PDF file itself.

**Comment by Tyrben** [comment]
Yeah, but if I substitute only the jpeg2000 content (pure data) by another from GIMP and just accord the size of the PDF generated by this operation (so that the PDF offset are not disturbed). The PDF is readable in Acrobat.
This is what makes me deduce it comes from the data.
A better guess would be that I misuse openjpeg. My code is more the less what can be found everywhere.
https://cpp.hotexamples.com/fr/examples/-/-/opj_encode/cpp-opj_encode-function-examples.html

The really best would be to have better info from Acrobat. 
Another clue was that Ghostscript, years ago, already had such an issue, solved it. But the error message was quiet similar to the current one of Acrobat (did they just introduce ghostscript code in acrobat... ?)

The last thing is: it's NEW. The last Acrobat update was several weeks ago. The bug appeared then.

**Comment by Tyrben** [comment]
Complement: Related to the Acrobat Patch 2021.011.20039
https://acrobat.uservoice.com/forums/590923-acrobat-for-windows-and-mac/suggestions/44810083-problems-with-jpeg2000-in-pdf-since-version-2021-0

**Comment by ValZapod** [comment]
Does not happen on acrobat for android so an obvious bug.

**Comment by ValZapod** [comment]
Was the sample crafted? Will extract jp2 file from the pdf.
